File: data-structure/day-1/217-contains-duplicate/217-contains-duplicate.cs
/**
 * Runtime: 167 ms
 * Memory Usage: 52.6 MB
 * Link: https://leetcode.com/submissions/detail/742341562/
 * //-----------------------------------------------------------------------------
 *    1. Define the hashset
 *    2. Loop over the nums
 *    3.    If the hashset contains the nums[i] -> return true
 *    4.    Add nums[i] to hashset
 *    5. Return false;
 * //-----------------------------------------------------------------------------
 * @param {number[]} nums
 * @return {number[]}
 */
function containsDuplicate(nums: number[]): boolean {

        // 1. Define the hashset
        //Dictionary is faster then hashSet in c#
        var hashSet = new Dictionary<int, int>();

        // 2. Loop over the nums
        for(int i = 0; i < nums.Length; i++)
        {
            // 3. If the hashset contains the nums[i] -> return true
            // 4. Add nums[i] to hashset
            if(!hashSet.TryAdd(nums[i], 1))
            {
                return true;
            }
        }

        // 5. Return false;
        return false;
    }
}

File: arrays-and-hashing/217-contains-duplicate/217-contains-duplicate.js
/**
 * Runtime: 94 ms
 * Memory Usage: 50.6 MB
 * Link: https://leetcode.com/problems/contains-duplicate/submissions/839901572/
 * //-----------------------------------------------------------------------------
 *    1. Define hash set.
 *    2. Loop over all nums
 *    3.    If hash set has item, return true.
 *    4.    Add item to hash set.
 *    5. Return false.
 * //-----------------------------------------------------------------------------
 * /**
 *  * @param {number[]} nums
 *  * @return {boolean}
 *  */
var containsDuplicate = function(nums) {
    
    // 1. Define hash set.
    let set = new Set();
    
    // 2. Loop over all nums
    for(let i of nums) {
        
        // 3. If hash set has item, return false.
       if(set.has(i)){
           return false;
       }

        // 4. Add item to hash set.
        set.add(i);
   }
    
    // 5. Return false.
    return true;
}